pkg/krmtotf/fetchlivestate.go

```diff
@@ -55,12 +55,13 @@ func FetchLiveState(ctx context.Context, resource *Resource, provider *tfschema.
 
 }
 
-// SkipOrphanedCheck Special handling for KMSCryptoKey that still lives after its parent KMSKeyRing is deleted.
-// We can import the tf state directly from itself instead of sourcing for its parent.
+// ShouldResolveParentForDelete
+// Special handling for KMSCryptoKey that still lives after its parent KMSKeyRing is deleted.
+// We can import the tf state directly from its selfLink instead of sourcing for its parent.
 // More info in b/279485255#comment14
-func SkipOrphanedCheck(resource *Resource) bool {
+func ShouldResolveParentForDelete(resource *Resource) bool {
 	allowlist := []string{"KMSCryptoKey"}
-	return slices.Contains(allowlist, resource.Kind) || !hasEmptySelfLink(resource)
+	return !slices.Contains(allowlist, resource.Kind) || hasEmptySelfLink(resource)
```

Comment:
To negative the skip function, I simply reverted back to Ziyue's original implementation of ShouldResolveParentForDelete:
```
func shouldGetImportIDFromSelfForDelete(resource *Resource) bool {
	return resource.Kind == "KMSCryptoKey"
}
func ShouldResolveParentForDelete(resource *Resource) bool {
	return !shouldGetImportIDFromSelfForDelete(resource) || hasEmptySelfLink(resource)
}
```
So it returns `!slices.Contains(allowlist, resource.Kind) || hasEmptySelfLink(resource)` 

Seems like my previous skip condition: `return slices.Contains(allowlist, resource.Kind) || !hasEmptySelfLink(resource)` is inaccurate.

Reply:
IIUC, the function current means:

If the kind is not KMSCryptoKey, return true;
If the kind is KMSCryptoKey, but it has the empty selfLink field, return true;
If the kind is KMSCryptoKey and it has the non-empty selfLink value, return false.

So it still resolves parent for delete even if it has non-empty selfLink value because it is not a KMSCryptoKey kind.

I think this is different from what the function comment said `We can import the tf state directly from its selfLink instead of sourcing for its parent.`?

Reply:
I get your point. And thanks for bring it up.
As I said I simply reverted back to the original implementation. the original implementation was only special handling for KMSCryptoKey resource. that's say, if a resource is not KMSCryptoKey, we still want to have complete checks on it, even we could simply delete it via its non-empty selfLink. The comment also mentioned that's special handling of KMSCryptoKey.
While, ideally, if a resource has non-empty selfLink, we could use selfLink as unique id and do not need to resolve its parent. 
What's your thought on this? Shall we keep this function limit to KMSCryptoKey, or expand it to all resource without selfLink?

Reply:
I personally think limit to KMSCryptoKey would have a minimal impact on our codebase. I am currently discussing with Justin whether we should delete via selfLink or a new "relevantResourceName" field.

Reply:
I think to minimize the impact, we should correct the comment to match the condition.

Further improvement can be done in the future.

Reply:
I revised the comment to be more specific to KMSCryptoKey.